src/service/data_products/taxa_count.py

```diff
@@ -232,11 +255,67 @@ async def get_taxa_counts(
     q = await _query(store, collection_id, load_ver, rank)
     for dp_proc in [dp_match, dp_sel]:
         await _add_subset_data_in_place(q, store, collection_id, load_ver, rank, dp_proc)
-        # For now always sort by the std data. See if ppl want sort by match/selection data
-        # before implementing something more sophisticated.
+
+    _sort_taxa_counts(q, sort_priority, [dp_match, dp_sel])
+
     return _taxa_counts(dp_match=dp_match, dp_sel=dp_sel, data=q)
 
 
+def _fill_missing_orders(sort_order: list[str], processed_count: list[str]):
+    # fill in missing orders with the default sort order if order exists in the processed count
+    if not isinstance(sort_order, list):
+        raise ValueError(f"sort_order must be a list of strings, provided: {sort_order}")
+
+    missing_orders = [order for order in _DEFAULT_SORT_ORDER
+                      if order not in sort_order and
+                      (order in processed_count or order == names.FLD_TAXA_COUNT_COUNT)]
+
+    return missing_orders + sort_order
+
+
+def _sort_taxa_counts(
+        q: list[dict[str, Any]],
+        sort_priority: str,
+        dp_list: list[models.DataProductProcess],
+        sort_by_count_already: bool = True):
```

Comment:
This isn't currently used - what's the use case for it?

Reply:
I intended to us it for situations where we found missing records linked to a match/selection and attach them to `q` causes the original sorting of `q` by count is no longer accurate.

Reply:
Ok, so it's a small optimization to avoid a resort when the match and select bars are already represented in the standard count bars, right?

Reply:
that's correct. If no new records are added, we don't necessarily need to resort by `count` since it's already sorted by `count`.

Reply:
:+1: